doc/dftb+/manual/dftbp.tex

```diff
@@ -4121,133 +4124,163 @@ \subsection{SSR22}
+  description of transition dipole moment between the electronic states in SA-REKS or SI-SA-REKS.
 
-  \item[\is{CGmaxIter}] Specifies the maximum number of iterations used in the conjugate gradient
-    based method. In general, the value of 20 is enough to solve the CP-REKS equations.
+\item[\is{Gradient}] Choice of gradient solver used in CP-REKS equations. This \is{Gradient} block has the following choices:
 
-  \item[\is{GradientTolerance}] Specifies the tolerance used in the conjugate gradient based method.
-    The \is{CGmaxIter} and \is{GradientTolerance} options are used when \is{GradientLevel} is 1 or 2.
+  \begin{description} 
+  \item[\is{ConjugateGradient}] Uses a congugate-gradient based algorithm. This
+    algorithm is usually recommended since it is considerably faster than other algorithms.
+    
+    \begin{ptable}
+      \kw{CGmaxIter} & i & & 20 & \\
+      \kw{Tolerance} & r & & 1e-8 & \\
+      \kw{Preconditioner} & l & & No & \\
+      \kw{SaveMemory} & l & & No & \\
+    \end{ptable}
 
-  \item[\is{RelaxedDensity}] If set to \is{Yes}, the file relaxed\_charge.dat is created. This file contains
-    a description of relaxed charges for \is{TargetState} or \is{TargetStateL}. The relaxed charges can
-    be used with external point charges in QM/MM calculations.
+    \begin{description}
+    \item[\is{CGmaxIter}] Specifies the maximum number of iterations used in the conjugate-gradient
+    based algorithm. In general, the value of 20 is enough to solve the CP-REKS equations.
 
-  \item[\is{NonAdiabaticCoupling}] If set to \is{Yes}, the nonadiabatic couplings between SI-SA-REKS
-    states are calculated. This option cannot be used in single-state REKS or SA-REKS state.
+    \item[\is{Tolerance}] Specifies the tolerance used in the conjugate-gradient based algorithm.
 
-  \item[\is{PrintLevel}] Specifies the printing level in standard output. This option determines the output up to
-    energy information (\is{PrintLevel} = 0), gradient information (\is{PrintLevel} = 1), detailed information about
-    SCC cycle and timing in gradient calculation (\is{PrintLevel} = 2).
+    \item[\is{Preconditioner}] If set to \is{Yes}, it uses a preconditiner in the conjugate-gradient based algorithm. In general,
+      the convergence speed is increased when this option sets to \is{Yes}, thus this option is recommended.
 
-  \item[\is{MemoryLevel}] Specifies the memory allocation of the variables used in CP-REKS equations.
-    First option shows fast computation speed but it requires large memory allocation, increasing as
-    $O(N_\mathrm{basis}^4)$. Second option shows relatively slow computation speed but it directly calculate
-    the CP-REKS variables without saving the variables. In general, the second option is recommended for large systems.
+    \item[\is{SaveMemory}] If set to \is{Yes}, some variables (A orbital hessian matrix and H-xc kernel) which need large
```

Comment:
"(an orbital hessian matrix and the H-xc kernel)"

Reply:
Done with commit 65d0cb5.